### DISCUSSION TITLE
renovate/artifacts pipeline always fails

### DISCUSSION THREAD

**Discussion by moeinio** [renovatebot/renovate#16963] [Request Help]
After setting up renovate, for some reason there is an external job in the pipeline that always fails, it does not have any logs and i am not sure why it is there.

I am using a self hosted repository for renovate bot in gitlab.com.

I have the same problem as already described with screenshots in the issue here:

[https://github.com/renovatebot/renovate/discussions/12559](url)

and here:

[https://github.com/renovatebot/config-help/issues/838](url)

Setting `trustLevel` to `high` seems to have solved the problem there.
I have done the equivalent to that by setting `allowCustomCrateRegistries` , `allowScripts` , and `exposeAllEnv` to `true`.  but it did it not work for me.

When this job fails, then the pipeline fails and it is a problem for us.

So the question is what exactly is this job trying to do and how can i understand why it fails?

**Comment by rarkins** [comment]
It's a status check and intended to fail. It indicates that artifacts updating failed and you should not merge the PR directly. 

This status check is an indication of a failure and is not the failure itself. You should solve the reason why the artifacts fail to update

**Reply by moeinio** [reply]
is there anyway i can find out why exactly the artefacts fail to update specifically for this job? they are fine for the rest of my pipeline.
I already set `LOG_LEVEL` to `debug`. Renovate logs seem to be fine, no errors.

**Reply by rarkins** [reply]
You need to capture the logs at the time that the status check was created. You won't see the errors printed in subsequent runs. You can get them to try again by ticking the retry/rebase checkbox in a MR

**Reply by moeinio** [reply]
the problem is a bit more clear now, renovate is unable to look up a package from a private npm registry.
I am really struggling to configure renovate to use private npm registry and their tokens, i looked up a lot of similar issues already.
is there a way to make renovate use .npmrc from gitlab environment variables? i don't want to hard code it (even  the encrypted string) into my configuration.

**Reply by rarkins** [reply]
Should be possible if you set it to RENOVATE_NPMRC

**Reply by moeinio** [reply]
Thank you for the fast replies. 
I have managed to solve the problem using `RENOVATE_NPMRC` as environment variable for renovate bot gitlab project.
I deleted all `hostRules` from my config and was able to Point `@myorg` scoped packages to our NPM registry using npmrc.
